analytics/pubstack/pubstack_module.go

```diff
@@ -0,0 +1,246 @@
+package pubstack
+
+import (
+	"fmt"
+	"github.com/prebid/prebid-server/analytics/pubstack/eventchannel"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/prebid/prebid-server/analytics/pubstack/helpers"
+
+	"github.com/prebid/prebid-server/analytics"
+)
+
+type Configuration struct {
+	ScopeId  string          `json:"scopeId"`
+	Endpoint string          `json:"endpoint"`
+	Features map[string]bool `json:"features"`
+}
+
+// routes for events
+const (
+	auction    = "auction"
+	cookieSync = "cookiesync"
+	amp        = "amp"
+	setUID     = "setuid"
+	video      = "video"
+)
+
+type bufferConfig struct {
+	timeout time.Duration
+	count   int64
+	size    int64
+}
+
+type PubstackModule struct {
+	eventChannels map[string]*eventchannel.EventChannel
+	configCh      chan *Configuration
+	scope         string
+	cfg           *Configuration
+	buffsCfg      *bufferConfig
+	mux           sync.Mutex
```

Comment:
Is this mux protecting the cfg object? If so, consider changing this to to a RWMutex since the reads will greatly outweigh the writes.

Reply:
I didn't know it, it's perfect. I also used it for the second mutex which it's the same pattern (N Readers- 1 Writer)